### PR TITLE
m: make Page a method

### PR DESCRIPTION
**Pull request by 89z**
`Page` is a great function, but it is limited, because it makes too many choices for the user. The global `DeInit` was a good start, but really that shouldve always been a `Pager` field.

Move `DeInit` inside `Pager`, and set default to `true` as before in the `NewPager` function. Then, users can call `Page`, but still have control over `DeInit`, as well as Line Numbers, and not have to worry about creating a Screen. Example code:

~~~go
r := m.NewReaderFromText("moar", strings.Repeat("moar\n", 99))
p := m.NewPager(r)
p.DeInit, p.ShowLineNumbers = false, false
p.Page()
~~~

### CODE REVIEW

**Comment by walles** [comment]
There's an embedding example in `README.md` that needs to be updated as well.

File: m/pager.go [hunk 1]
```diff
@@ -44,6 +44,10 @@ type Pager struct {
 
 	// NewPager shows lines by default, this field can hide them
 	ShowLineNumbers bool
+
+	// If true, pager will clear the screen on return. If false, pager will
+	// clear the last line, and show the cursor.
+	DeInit bool
```
Comment (walles) [inline m/pager.go:50]: `DeInit` reflects what it does internally, maybe something like `RestoreScreenWhenDone` would reflect better what effect a user would see from changing it?

Not sure what a great name would be, if you don't want to change it I'm fine with `DeInit` for now.

File: m/pager.go [hunk 2]
```diff
@@ -44,6 +44,10 @@ type Pager struct {
 
 	// NewPager shows lines by default, this field can hide them
 	ShowLineNumbers bool
+
+	// If true, pager will clear the screen on return. If false, pager will
+	// clear the last line, and show the cursor.
```
Comment (walles) [inline m/pager.go:49]: This comment is excellent.

**Comment by walles** [comment]
If you fix the `README.md`, and at least consider my inline naming comment this should be good enough to merge.

**Comment by 89z** [comment]
I updated the Readme. Regarding `DeInit`, I like it as is, but if you insist we can use the other option.

**Pull request merged**